### PR TITLE
vulnerability report updated. Setting and forms secured but broke site.

### PR DESCRIPTION
**Pull request by jaredscarr**
Totally fucking broken

### CODE REVIEW

**Comment by cktricky** [comment]
We're sort of guessing these are accidental commits meant for a CodeFellows repo? @jaredscarr

**Comment by jaredscarr** [comment]
Oh no. Sorry about that.

**Comment by cktricky** [comment]
No worries, tbh we're happy folks are getting some use out of it.

File: vulnerability-report.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+Reviewing nVisium Task Manager
+
+Reviewer: Jared Scarr
+Date: 4-25-2016
+
+Vulnerability 1
```
Comment (nhuntwalker) [inline vulnerability-report.md]: And the name of vulnerability 1 is.....?

File: vulnerability-report.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Exposure
+
+The file upload field is being executed directly to the database. Giving the user
```
Comment (nhuntwalker) [inline vulnerability-report.md]: The file upload field on which page? Or pages?

File: vulnerability-report.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+ replaces:
+
+ curs = connection.cursor()
+curs.execute(
+    "insert into taskManager_file ('name','path','project_id') 
+     values ('%s','%s',%s)" %
+     (name, upload_path, project_id))
+
+Vulnerability 2
```
Comment (nhuntwalker) [inline vulnerability-report.md]: Obligatory comment...

File: vulnerability-report.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+Reviewing nVisium Task Manager
+
+Reviewer: Jared Scarr
```
Comment (nhuntwalker) [inline vulnerability-report.md]: No partner?

File: vulnerability-report.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+     (name, upload_path, project_id))
+
+Vulnerability 2
+
+Exposure
+
+The registration view is vulnerable because it includes all fields. Including 
+the ability to be a super user. A little change in the html and the user may
+manually enable this prior to registration to give themselves superuser status.
+You can see the cookie right there. It can be read my anybody.
```
Comment (nhuntwalker) [inline vulnerability-report.md]: And a superuser would be able to do....?

File: vulnerability-report.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+forms.py and settings.py in the taskManager app contained the vulnerability. I was able to fix
+the the code with the following adjustment(s):
+
+forms.py
+
+on line 75 
+exclude = ['groups', 'user_permissions', 'last_login', 'date_joined', 'is_active']
+
+should be changed to:
+include = ['username', 'first_name', 'last_name', 'password']
```
Comment (nhuntwalker) [inline vulnerability-report.md]: or just `fields = ['username', 'first_name', 'last_name', 'password']`

File: vulnerability-report.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+Line 84 should be removed and a cookie session engine configured. Add 'django.contrib.sessions' to your INSTALLED_APPS.
+
+SESSION_ENGINE=("django.contrib.sessions.backends.signed_cookies")
+SESSION_COOKIE_HTTPONLY=True
+
+Vulnerablility 3
+
+Exposure
+
+The user update user profile template is vulnerable to XXS.
```
Comment (nhuntwalker) [inline vulnerability-report.md]: How? Why? Where was the exposure on the site? How did you exploit that vulnerability? Also, it's XSS (Cross-Site Scripting).

File: vulnerability-report.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+Repair
+
+templates/profile.html
+
```
Comment (nhuntwalker) [inline vulnerability-report.md]: A CSRF token isn't going to solve the cross-site scripting problem. That solves the CSRF problem. Keep trying.

File: vulnerability-report.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+needs a csrf token ({% csrf token %}) added to the form. Really any form that
+is going to take user input.
+
+Vulnerability 4
+
+Exposure
+
+You can see all task no matter if they belong to the user signed in or not. 
+You can type an id into the request and if it exists you have access.
```
Comment (nhuntwalker) [inline vulnerability-report.md]: Provide an example url.

**Pull request closed without merging**